File: src/utils/env.js
import fs from "node:fs";
import path from "node:path";
import process from "node:process";

function loadSingleEnvFile(filePath) {
  if (!fs.existsSync(filePath)) {
    return;
  }

  const content = fs.readFileSync(filePath, "utf8");
  const lines = content.split(/\r?\n/);

  for (const line of lines) {
    const trimmed = line.trim();

    if (!trimmed || trimmed.startsWith("#")) {
      continue;
    }

    const separatorIndex = trimmed.indexOf("=");

    if (separatorIndex === -1) {
      continue;
    }

    const key = trimmed.slice(0, separatorIndex).trim();
    const rawValue = trimmed.slice(separatorIndex + 1).trim();
    const value = rawValue.replace(/^['"]|['"]$/g, "");

    if (!(key in process.env)) {
      process.env[key] = value;
    }
  }
}

export function loadEnvFile() {
  const cwd = process.cwd();
  const explicitPath = process.env.CONTEXT_HUB_ENV_FILE;

  if (explicitPath) {
    loadSingleEnvFile(path.resolve(explicitPath));
    return;
  }

  const candidates = [".env.local", ".env"].map((name) => path.join(cwd, name));

  for (const candidate of candidates) {
    loadSingleEnvFile(candidate);
  }
}

File: contexthub/env.py
from __future__ import annotations

import os
from pathlib import Path


def _load_single_env_file(file_path: Path) -> None:
    if not file_path.exists():
        return

    for raw_line in file_path.read_text(encoding="utf-8").splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue

        key, value = line.split("=", 1)
        key = key.strip()
        value = value.strip().strip("\"'")
        os.environ.setdefault(key, value)


def load_env_files() -> None:
    explicit_path = os.environ.get("CONTEXT_HUB_ENV_FILE")
    if explicit_path:
        _load_single_env_file(Path(explicit_path).expanduser().resolve())
        return

    root = Path.cwd()
    for candidate in (root / ".env.local", root / ".env"):
        _load_single_env_file(candidate)
